controllers/application_poller_controller.go

```diff
@@ -89,15 +100,21 @@ func (p *Poller) Start(ch <-chan struct{}) error {
 		return nil
 	}
 
-	activityCh := make(chan applications.ActivityItem)
-	go subscriber.Subscribe(ctx, activityCh)
-
 	for {
-		select {
-		case <-ch:
-			return nil
-		case activity := <-activityCh:
-			p.handleActivity(ctx, activity)
+		activityCh := make(chan applications.ActivityItem)
+
+		go func() {
+			for activityItem := range activityCh {
+				p.handleActivity(ctx, activityItem)
+			}
+		}()
+
+		if err := subscriber.Subscribe(ctx, activityCh); err != nil {
+			// Clean exit/shutdown
+			if errors.Is(err, context.Canceled) {
+				return nil
+			}
+			p.log.Error(err, "Application service connection interrupted, restarting")
```

Comment:
Do we need to `return nil` if `err == nil`? I don't think it's possible with the current code, but it might be unexpected at a later time.